test/gtest/multimarginloss.hpp

```diff
@@ -0,0 +1,236 @@
+            std::fill(workspace.begin(), workspace.end(), 0);
+            workspace_dev = handle.Write(workspace.data);
+        }
+    }
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        miopenStatus_t status;
+        if(reduction_mode == MIOPEN_LOSS_REDUCTION_NONE)
+        {
+            cpu_multimarginloss_unreduced_forward<T>(input, target, weight, ref_output, p, margin);
+
+            status = miopen::MultiMarginLossUnreducedForward(handle,
+                                                             input.desc,
+                                                             input_dev.get(),
+                                                             target.desc,
+                                                             target_dev.get(),
+                                                             weight.desc,
+                                                             weight_dev.get(),
+                                                             output.desc,
+                                                             output_dev.get(),
+                                                             p,
+                                                             margin);
+        }
+        else
+        {
+            cpu_multimarginloss_reduced_forward<T>(
+                input,
+                target,
+                weight,
+                ref_output,
+                p,
+                margin,
+                (reduction_mode == MIOPEN_LOSS_REDUCTION_MEAN) ? input.desc.GetLengths()[0] : 1);
+
+            status = miopen::MultiMarginLossForward(handle,
+                                                    workspace_dev.get(),
+                                                    ws_sizeInBytes,
+                                                    input.desc,
+                                                    input_dev.get(),
+                                                    target.desc,
+                                                    target_dev.get(),
+                                                    weight.desc,
+                                                    weight_dev.get(),
+                                                    output.desc,
+                                                    output_dev.get(),
+                                                    p,
+                                                    margin,
+                                                    reduction_mode);
+        }
+        EXPECT_EQ(status, miopenStatusSuccess);
+
+        // Write from GPU to CPU
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        // Computation error of fp16 is ~2^13 (=8192) bigger than
+        // the one of fp32 because mantissa is shorter by 13 bits.
+        auto threshold = std::is_same<T, float>::value ? 1.5e-6 : 8.2e-3;
```

Comment:
Rename `threshold` to `tolerance`. For being similar to MIOpenDriver.

Reply:
fixed it